src/SiteMaster/Core/Auditor/Scan/Progress.php

```diff
@@ -61,14 +61,8 @@ public function getQueuePosition()
             return false;
         }
 
-        $site = $this->scan->getSite();
-        
-        if (!$homepage = Page::getByScanIDAndURI($this->scan->id, $site->base_url)) {
-            return false;
-        }
-
         $queue = new Queued(array('limit'=>-1));
         
-        return array_search($homepage->id, $queue->getInnerIterator()->getArrayCopy());
+        return array_search($this->scan->id, $queue->getInnerIterator()->getArrayCopy());
```

Comment:
This logic make more sense being a part of the `Queued` class to keep business logic closer to the data responsibility.